### PR TITLE
Khalil/unicast authorized sender v0.27

### PR DESCRIPTION
**Pull request by kc1116**
This PR backports changes from https://github.com/onflow/flow-go/pull/2832

### CODE REVIEW

File: cmd/observer/node_builder/observer_builder.go [hunk 1]
```diff
@@ -69,9 +71,7 @@ import (
 	bstorage "github.com/onflow/flow-go/storage/badger"
 	"github.com/onflow/flow-go/utils/grpcutils"
 	"github.com/onflow/flow-go/utils/io"
-
-	"github.com/rs/zerolog"
-	"github.com/spf13/pflag"
+	accessproto "github.com/onflow/flow/protobuf/go/flow/access"
```
Comment (peterargue) [inline cmd/observer/node_builder/observer_builder.go:74]: this will probably flag a linter error

File: module/mock/hot_stuff_follower.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.13.1. DO NOT EDIT.
```
Comment (peterargue) [inline module/mock/hot_stuff_follower.go:1]: changes to this file weren't in the original

Comment (kc1116) [reply module/mock/hot_stuff_follower.go]: https://github.com/onflow/flow-go/pull/3005/commits/1eb538b3a3fd7f9e77d8c06a65ff3e3bf5be9fda

File: network/test/unicast_authorization_test.go [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/network/mocknetwork"
+	"github.com/onflow/flow-go/network/slashing"
+	"github.com/onflow/flow-go/network/validator"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
```
Comment (peterargue) [inline network/test/unicast_authorization_test.go:25]: these will cause linter errors

**Review by peterargue — APPROVED** [review]
flagged a few minor differences from the original PR, but otherwise look good!

**Review by yhassanzadeh13 — APPROVED** [review]

**Pull request merged**